nuitka/tools/testing/OutputComparison.py

```diff
@@ -202,7 +226,9 @@ def compareOutput(
     diff = difflib.unified_diff(
         makeDiffable(out_cpython, ignore_warnings, ignore_infos, syntax_errors),
         makeDiffable(out_nuitka, ignore_warnings, ignore_infos, syntax_errors),
-        "{program} ({detail})".format(program=os.environ["PYTHON"], detail=kind),
+        "{program} ({detail})".format(
```

Comment:
One important thing, *never* add defaults where you lack the knowledge to invent them. In this function, how can you know what has been used, it may not be `python` at all.

Luckily, and not by chance, there is no way that this is needed, after calling setup, which every test suite does, it will be set. I think you didn't do that initially, which broke assumptions like this. This should be removable.